### PR TITLE
interpreters/lua: replace sourceVersion with lib.versions

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by teto — APPROVED** [review]

File: pkgs/development/interpreters/luajit/2.0.nix [hunk 1]
```diff
@@ -1,13 +1,17 @@
-{ self, callPackage, lib, passthruFun }:
+{ self, callPackage, fetchFromGitHub, lib, passthruFun }:
+
 callPackage ./default.nix {
-  sourceVersion = { major = "2"; minor = "0"; patch = "5"; };
-  inherit self passthruFun;
   version = "2.0.5-2022-09-13";
-  rev = "46e62cd963a426e83a60f691dcbbeb742c7b3ba2";
-  isStable = true;
```
Comment (Artturin) [inline pkgs/development/interpreters/luajit/2.0.nix:7]: eval fails because isStable was removed

Comment (SuperSandro2000) [reply pkgs/development/interpreters/luajit/2.0.nix:7]: dropped that in another branch

File: pkgs/development/interpreters/luajit/2.0.nix [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-  hash = "sha256-/XR9+6NjXs2TrUVKJNkH2h970BkDNFqMDJTWcy/bswU=";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/luajit/2.0.nix:5]:
```suggestion
  isStable = true;
  src = fetchFromGitHub {
```

File: pkgs/development/interpreters/luajit/2.1.nix [hunk 3]
```diff
@@ -1,9 +1,12 @@
-{ self, callPackage, passthruFun }:
+{ self, callPackage, fetchFromGitHub, passthruFun }:
 callPackage ./default.nix {
-  sourceVersion = { major = "2"; minor = "1"; patch = "0"; };
-  inherit self passthruFun;
   version = "2.1.0-2022-10-04";
-  rev = "6c4826f12c4d33b8b978004bc681eb1eef2be977";
-  isStable = false;
-  hash = "sha256-GMgoSVHrfIuLdk8mW9XgdemNFsAkkQR4wiGGjaAXAKg=";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/luajit/2.1.nix:4]:
```suggestion
  isStable = false;
  src = fetchFromGitHub {
```

**Comment by SuperSandro2000** [comment]
Mostly vim plugin rebuilds. We could keep targeting master. This would also reduce merge conflicts with the other PR I have open.

File: pkgs/development/interpreters/lua-5/build-lua-package.nix [hunk 4]
```diff
@@ -14,7 +14,7 @@
 , rockspecVersion ? version
 
 # by default prefix `name` e.g. "lua5.2-${name}"
-, namePrefix ? "${lua.pname}${lua.sourceVersion.major}.${lua.sourceVersion.minor}-"
+, namePrefix ? "${lua.pname}${lib.versions.majorMinor version}-"
```
Comment (vcunat) [inline pkgs/development/interpreters/lua-5/build-lua-package.nix:17]: This was a mistake.  See: https://github.com/NixOS/nixpkgs/pull/223349

**Pull request merged**